roles/openshift_master_upgrade/tasks/restart_services.yml

```diff
@@ -0,0 +1,31 @@
+---
+# input variables:
+# - openshift.common.service_type
+# - openshift_master_ha
+# - openshift.common.hostname
+
+- name: Restart master
+  service:
+    name: "{{ openshift.common.service_type }}-master"
+    state: restarted
+  when: not openshift_master_ha | bool
+- name: Restart master API
+  service:
+    name: "{{ openshift.common.service_type }}-master-api"
+    state: restarted
+  when: openshift_master_ha | bool
+- name: Wait for master API to come back online
+  wait_for:
+    host: "{{ openshift.common.hostname }}"
+    state: started
+    delay: 10
+    port: "{{ openshift.master.api_port }}"
```

Comment:
I'm wondering if a do/until loop with the uri module testing for 'healthz/ready' would be better than wait_for here.